gutenberg.php

```diff
@@ -74,3 +74,18 @@ function gutenberg_pre_init() {
 
 	require_once __DIR__ . '/lib/load.php';
 }
+
+if ( ! function_exists( 'filter_query_block_exclude_current' ) ) {
+	/**
+	 * This is included for demonstration purposes only.
+	 * We should make this change in the core `build_query_vars_from_query_block` function.
+	 */
+	function filter_query_block_exclude_current( $query, $block ) {
+		if ( isset( $block->context['query']['excludeCurrent'] ) && $block->context['query']['excludeCurrent'] ) {
+			$query['post__not_in'] = array( get_the_ID() );
+		}
+
+		return $query;
+	}
+	add_filter( 'query_loop_block_query_vars', 'filter_query_block_exclude_current', 10, 2 );
+}
```

Comment:
Temporary changes to Gutenberg, that will be removed once the change is made in core and the plugin only supports that WordPress version or newer, should be added to the lib/compat/version folder.
You can find examples of this for reference in the wordpress-6.7 folder.

Reply:
Once approved, a Trac ticket and pull request needs to be created, referencing the Gutenberg PR, and a changelog entry needs to be added:
https://github.com/WordPress/gutenberg/tree/trunk/backport-changelog

Reply:
@carolinan is right about the location of such code. Additionally I don't think the filter would work in this case, because Query Loop internally can also set the `post_not_in` value ([check here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/blocks.php#L2370)).

So we would probably need to override this function in GB. It's been a while that we had changes there, but I had implemented some code in the webpack config to use `build_query_vars_from_query_block` with `gutenberg` prefix if available. We don't currently have an override for this function in GB trunk.

Check this for [webpack](https://github.com/WordPress/gutenberg/blob/trunk/tools/webpack/blocks.js#L32).